src/SparkFun_AS7331.h

```diff
@@ -14,1312 +14,716 @@
+
+    uint8_t _breakTime; // Local config value. Value is in us/8. EX: _breakTime = 20 means 20*8 = 160us.
+    uint8_t _numEdges;  // Local config value. Edges seen on SYN pin before ending conversion in SYND mode.
+
+    bool _readyPinMode;         // Local config value. False is Push/Pull True is Open/Drain.
+    bool _dividerEnabled;       // Local config value. False is disabled, True is enabled.
+    bool _tempConvEnabled;      // Local config value. False is disabled, True is enabled.
+    bool _indexMode;            // Local config value. False is for controllers without repeat start.
+    bool _standbyState;         // Local state value. False means the device is not in standby mode.
+    bool _startState;           // Local state value. False means the device is not allowed to measure.
+    bool _powerDownEnableState; // Local state value. False means the device is NOT POWERED DOWN. This is inverse.
+
+    as7331_dev_op_state_t _opMode; // Local state value. Configuration or Measure operating mode.
+    as7331_gain_t _sensorGain;     // Local config value. Sensor gain stored as (1 << (11 - gain)).
+    as7331_conv_clk_freq_t _cclk;  // Local config value. Sensor's conversion clock stored as 1024*(1 << cclk) kHz.
+    as7331_meas_mode_t _mmode; // Local state value. Details the device's measurement mode as CONT, CMD, SYNS, or SYND.
+    as7331_conv_time_t _conversionTime; // Local config value. Contains the conversion time stored as (1 << time) ms.
+    as7331_divider_val_t _dividerRange; // Local config value. Contains the predivider value stored as (1 << 1 + range).
+
+    float _uva;         // Last valid UVA result in uW/cm2.
+    float _uvb;         // Last valid UVB result in uW/cm2.
+    float _uvc;         // Last valid UVC result in uW/cm2.
+    float _temperature; // Last valid temperature result in degC.
+
+    uint32_t _outputConversionTime; // Last valid output conversion time result in of clock cycle count.
+
+    float _conversionA; // Internal conversion factor for the UVA sensor.
+    float _conversionB; // Internal conversion factor for the UVB sensor.
+    float _conversionC; // Internal conversion factor for the UVC sensor.
+
+    const float _fsrA = 348160.0; // Full Scale Resolution for the UVA sensor.
+    const float _fsrB = 387072.0; // Full Scale Resolution for the UVB sensor.
+    const float _fsrC = 169984.0; // Full Scale Resolution for the UVC sensor.
+};
 
-            sfe_as7331_reg_cfg_optreg_t optreg;
+class SfeAS7331ArdI2C : public SfeAS7331Driver
+{
+  public:
+    SfeAS7331ArdI2C()
+    {
+    }
 
-            result = getOptIndex(&optreg);
-            if(SFE_BUS_OK != result)
-                return result;
+    /// @brief  Sets up Arduino I2C driver using the specified I2C address then calls the super class begin.
+    /// @param address Address of the I2C device.
+    /// @return True if successful, false otherwise.
+    bool begin(const uint8_t &address = kDefaultAS7331Addr)
```

Comment:
A lot of our Arduino libraries have a begin that takes a Wire reference - should we do this here (maybe have a couple begins . This is minor.

Reply:
I'd argue this is not minor, some platforms just won't work with the default `Wire` (the Uno R4 comes to mind, it needs `Wire1` for the Qwiic connector). Can just add a second parameter that defaults to `Wire` like so:

```
bool begin(const uint8_t &address = kDefaultAS7331Addr, TwoWire& wirePort)
{
    if (_theI2CBus.init(wirePort, address) != kSTkErrOk)
        return false;

    // etc...
}
```

This is pretty typical of `begin()` in our libraries

Reply:
That's easy enough to do. Just for clarification, did you mean to not include a default in the begin function?

```C++
bool begin(const uint8_t &address = kDefaultAS7331Addr, TwoWire &wirePort = Wire)
{
  ...
}
```

Or was this to be a second begin function?

```C++
bool begin(const uint8_t &address = kDefaultAS7331Addr)
{
  ...
}

bool begin(const uint8_t &address = kDefaultAS7331Addr, TwoWire &wirePort)
{
  ...
}
```

Reply:
I think I echo @sfe-SparkFro, the user exposed begin() has, for the last 4+ years followed the following format:

* bool begin() - Nothing needed
* bool begin(alternate device address)
* bool begin(alternate device address, alternate Wire port)

Bool is ideally returned so the user can quickly check if the device is setup and online. If a more complex result can be returned, a different function should be made available. Rarely does the user need an alternate device address. *Very* rarely does the user need to modify the Wire port, but it's super nice to have when you need it.

Reply:
Could that not be accomplished with defaults though?

```C++
bool begin(const uint8_t &address = kDefaultAS7331Addr, TwoWire &wirePort = Wire)
{
  ...
}
```

This could solve all 3 of those use cases. The only thing it wouldn't solve would be the user wanting to only set the Wire port but not the address, however I feel like that would be an extremely rare use case and the user can just enter the default address at that point.

Reply:
Yup, your solution works, I just want to be sure we don't change things.